src/tests/resolve.rs

```diff
@@ -151,6 +151,21 @@ fn resolve_dot() {
     }
 }
 
+#[test]
+fn symlink_with_nested_node_modules() {
+    let f = super::fixture_root().join("symlink-with-nested-node_modules");
+
+    let resolver = Resolver::default();
+    let resolved_path =
+        resolver.resolve(f.join("bar/node_modules/foo"), "dep").map(|r| r.full_path());
+    assert_eq!(resolved_path, Ok(f.join("foo/node_modules/dep/index.js")));
```

Comment:
@coderabbitai Please help to fix this failing test case by modifying `src/lib.rs`, the test case is correct, the current resolving function is buggy in these two cases.

Reply:
The second test case with `symlinks: false` option is still failing with this change.